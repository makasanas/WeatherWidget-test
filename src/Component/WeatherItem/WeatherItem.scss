.weather-item {
    text-align: center;
    margin: 0 5px;

    p {
        font-size: 14px;
        text-transform: uppercase;
    }
}