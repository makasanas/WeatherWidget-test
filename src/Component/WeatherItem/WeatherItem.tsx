import React, { FunctionComponent } from 'react';
import moment from 'moment';

import WeatherIcon from '../WeatherIcon/WeatherIcon';

import './WeatherItem.scss';

interface WeatherItemProps {
  weatherInfo: any;
}

const WeatherItem: FunctionComponent<WeatherItemProps> = props => {
  //Timestamp to date format i.e Wed
  const getDay = (time: number) => {
    return moment.unix(time).format('ddd');
  };

  return (
    <div className={'weather-item'}>
      <p>{getDay(props.weatherInfo.dt)}</p>
      <WeatherIcon icon={props.weatherInfo.weather[0].icon} />
      <p>
        {' '}
        {props.weatherInfo.main.temp}
        <span>&#8451;</span>
      </p>
    </div>
  );
};

export default WeatherItem;
