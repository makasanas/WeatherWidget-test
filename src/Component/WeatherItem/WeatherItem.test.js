import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import WeatherItem from './WeatherItem';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);


describe('WeatherItem', () => {
    test('should render app.js', () => {
        const store = mockStore({
            weather: {
                current: null,
                loading: false,
                error: null,
                initialLoad: true,
            }
        });
        const tree = render(
            <Provider store={store}>
                <WeatherItem weatherInfo={
                    {
                        "dt": 1487246400,
                        "main": {
                            "temp": 286.67,
                            "temp_min": 281.556,
                            "temp_max": 286.67,
                            "pressure": 972.73,
                            "sea_level": 1046.46,
                            "grnd_level": 972.73,
                            "humidity": 75,
                            "temp_kf": 5.11
                        },
                        "weather": [
                            {
                                "id": 800,
                                "main": "Clear",
                                "description": "clear sky",
                                "icon": "01d"
                            }
                        ],
                    }
                } />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });

})

