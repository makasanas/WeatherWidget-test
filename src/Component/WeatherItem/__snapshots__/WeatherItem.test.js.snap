// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem should render app.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="weather-item"
      >
        <p>
          Thu
        </p>
        <div
          class="weather-img"
        >
          <img
            alt="wthr img"
            src="http://openweathermap.org/img/wn/01d.png"
          />
        </div>
        <p>
           
          286.67
          <span>
            ℃
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weather-item"
    >
      <p>
        Thu
      </p>
      <div
        class="weather-img"
      >
        <img
          alt="wthr img"
          src="http://openweathermap.org/img/wn/01d.png"
        />
      </div>
      <p>
         
        286.67
        <span>
          ℃
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
